h2o-core/src/test/java/hex/pipeline/PipelineTest.java

```diff
@@ -0,0 +1,495 @@
+package hex.pipeline;
+
+import hex.Model;
+import hex.ModelBuilder;
+import hex.pipeline.DataTransformerTest.AddDummyCVColumnTransformer;
+import hex.pipeline.DataTransformerTest.AddRandomColumnTransformer;
+import hex.pipeline.DataTransformerTest.FrameCheckerAsTransformer;
+import hex.pipeline.DataTransformerTest.FrameTrackerAsTransformer;
+import hex.pipeline.PipelineModel.PipelineOutput;
+import hex.pipeline.PipelineModel.PipelineParameters;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import water.*;
+import water.fvec.Frame;
+import water.fvec.TestFrameBuilder;
+import water.fvec.Vec;
+import water.junit.rules.DKVIsolation;
+import water.junit.rules.ScopeTracker;
+import water.runner.CloudSize;
+import water.runner.H2ORunner;
+import water.test.dummy.DummyModel;
+import water.test.dummy.DummyModelParameters;
+import water.util.ArrayUtils;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.*;
+import static water.TestUtil.*;
+
+@RunWith(H2ORunner.class)
+@CloudSize(1)
+public class PipelineTest {
+
+  @Rule
+  public ScopeTracker scope = new ScopeTracker();
+
+  @Rule
+  public DKVIsolation isolation = new DKVIsolation();
+
+  private void checkFrameState(Frame fr) {
+    assertNotNull(fr.getKey());
+    assertNotNull(DKV.get(fr.getKey()));
+    assertFrameEquals(fr, DKV.getGet(fr.getKey()), 1e-10);
+    for (int i=0; i<fr.keys().length; i++) {
+      Key<Vec> k = fr.keys()[i];
+      assertNotNull(k);
+      assertNotNull(k.get());
+      assertVecEquals(fr.vec(i), k.get(), 1e-10);
+    }
+  }
```

Comment:
can you add an example with pipeline say multiply a column by 2 and then build a GLM or GBM or DRF model?

Reply:
@wendycwong added a quick `MultiplyNumericColumnTransformer` for the test below, with assertions verifying that it's applied correctly and that the original frame is not modified.